### PR TITLE
ADE SP March QoS

### PR DESCRIPTION
**Pull request by vimish**
1. Fail OS encryption if memory less than 7 GB
2. Do apt-get update only when install fails

### CODE REVIEW

File: VMEncryption/main/handle.py [hunk 1]
```diff
@@ -628,6 +628,16 @@ def enable_encryption():
                        level=CommonVariables.WarningLevel)
             hutil.redo_last_status()
             exit_without_status_report()
+    
+    cutil = CheckUtil(logger)
```
Comment (vermashi) [inline VMEncryption/main/handle.py]: Please add this check to the fatal checks method in CheckUtil instead.

File: VMEncryption/main/patch/UbuntuPatching.py [hunk 2]
```diff
@@ -58,12 +58,17 @@ def __init__(self, logger, distro_info):
...
-        self.command_executor.Execute(cmd)
-        
         packages = ['cryptsetup-bin']
         cmd = " ".join(['apt-get', 'install', '-y'] + packages)
-        self.command_executor.Execute(cmd)
+        return_code = self.command_executor.Execute(cmd)
+        
+        # If install fails, try running apt-get update and then try install again
+        if return_code != 0:
+            cmd = " ".join(['apt-get', '-o Acquire::ForceIPv4=true', '-y', 'update'])
```
Comment (vermashi) [inline VMEncryption/main/patch/UbuntuPatching.py]: Why are you writing out the command like this?
Just do it normally, its more readable: "apt-get -o Acquire::ForceIPv4-True -y update"

File: VMEncryption/main/handle.py (hunk 1 above)
Comment (vermashi) [reply VMEncryption/main/handle.py]: I'm talking about the method already called at line 534 of this file.

Comment (vermashi) [reply VMEncryption/main/handle.py]: if (volume_type is OS/ALL and OSdiskStatus is NOtEncrypted (ref: line 117) and operation is EnableType (ref: line 141) )
    do the memory check ()

File: VMEncryption/main/handle.py [hunk 3]
```diff
@@ -528,10 +528,25 @@ def enable():
...
+        disk_util = DiskUtil(hutil=hutil, patching=DistroPatcher, logger=logger, encryption_environment=encryption_environment)
+        bek_util = BekUtil(disk_util, logger)
+        existing_passphrase_file = None
+        encryption_config = EncryptionConfig(encryption_environment=encryption_environment, logger=logger)
+
+        existing_passphrase_file = bek_util.get_bek_passphrase_file(encryption_config)
+        if existing_passphrase_file is not None:
+            disk_util.consolidate_azure_crypt_mount(existing_passphrase_file)
+            mount_encrypted_disks(disk_util=disk_util,
+                                bek_util=bek_util,
```
Comment (vermashi) [inline VMEncryption/main/handle.py]: Fix indentation here

**Comment by vermashi** [comment]
Looks good but look at comments

**Comment by vimish** [comment]
@vermashi @ejarvi @mayank88mahajan 
Addressed review comments.

**Pull request merged**